src/maliput_malidrive/base/road_geometry.cc

```diff
@@ -342,26 +400,55 @@ maliput::api::RoadPosition RoadGeometry::OpenScenarioRoadPositionToMaliputRoadPo
     const double r_road_curve_max = r_road_curve + width / 2.;
     // r-coordinates are located in RoadCurve FRAME while t-coordinate are respect to Road Reference Line in the ground.
     const double roll_at_p = target_segment->road_curve()->superelevation()->f(p);
-    const double xodr_t_projected_on_r = xodr_t / std::cos(roll_at_p);
+    const double xodr_t_projected_on_r = xodr_road_position.t / std::cos(roll_at_p);
     if (is_greater_than_or_close(xodr_t_projected_on_r, r_road_curve_min) &&
         is_less_than_or_close(xodr_t_projected_on_r, r_road_curve_max)) {
       // If the lane r value lies the limit of lanes then the right lane is the target lane as lanes are iterated from
       // right to left by definition.
       target_lane = mali_lane;
       r = mali_lane->to_lane_r(p, xodr_t_projected_on_r);
-      mali_lane_s = target_lane->LaneSFromTrackS(xodr_s);
+      mali_lane_s = target_lane->LaneSFromTrackS(xodr_road_position.s);
       break;
     }
   }
   if (target_lane == nullptr) {
     MALIDRIVE_THROW_MESSAGE(
         "A maliput lane can't be found for the given OpenSCENARIO road position: "
         "RoadID: " +
-        std::to_string(xodr_road_id) + ", s: " + std::to_string(xodr_s) + ", t: " + std::to_string(xodr_t));
+        std::to_string(xodr_road_position.road_id) + ", s: " + std::to_string(xodr_road_position.s) +
+        ", t: " + std::to_string(xodr_road_position.t));
   }
   return maliput::api::RoadPosition{target_lane, maliput::api::LanePosition{mali_lane_s, r, 0.}};
 }
 
+RoadGeometry::OpenScenarioRoadPosition RoadGeometry::MaliputRoadPositionToOpenScenarioRoadPosition(
+    const maliput::api::RoadPosition& road_position) const {
+  OpenScenarioRoadPosition xodr_road_position;
+  const auto mali_lane = dynamic_cast<const Lane*>(road_position.lane);
+  xodr_road_position.road_id = mali_lane->get_track();
+  xodr_road_position.s = mali_lane->TrackSFromLaneS(road_position.pos.s());
+  const auto segment = dynamic_cast<const Segment*>(mali_lane->segment());
+  double t = 0.;
+
+  int t_direction = mali_lane->get_lane_id() > 0 ? 1 : -1;
+  int n_lanes = segment->num_lanes();
+  // Start at 1 to ignore lane 0.
+  for (int i = 1; i < std::abs(mali_lane->get_lane_id()); ++i) {
+    int index = n_lanes / 2 + i * t_direction + (t_direction > 0 ? -1 : 0);
+    const Lane* curr_mali_lane = dynamic_cast<const Lane*>(segment->lane(index));
+    const double width = curr_mali_lane->lane_width_at(xodr_road_position.s);
+    const double roll_at_p = segment->road_curve()->superelevation()->f(xodr_road_position.s);
+    t += width * std::cos(roll_at_p) * t_direction;
+  }
```

Comment:
I think that you can minimize this, because within mali_lane you have the lane_offset function. (It is private but we can make it public if we need). The lane offset function is constructed with the information of the offset of previous lanes.

Reply:
In fact, if you receive the s r h you can use lane->to_reference_r (s, r) to get the `r` at the road curve level. With that and knowing the roll due to the superelevation you can get the final `t`. Right?

Reply:
Great suggestion! Done.